### ISSUE TITLE
`repeated-keyword` false-positive when overriding existing dict keys

### ISSUE THREAD

**Issue by alimuldal** [PyCQA/pylint#4861]
### Bug description

```python
This legitimate code produces a false-positive `repeated-keyword` finding on the final line:


def my_func(this, that):
    return this, that

kwargs = dict(this=1, that=2)
new_kwargs = dict(kwargs, that=3)
my_func(**new_kwargs)
```

### Command used

```shell
`pylint test_module.py`
```

### Pylint output

```shell
/tmp/test.py:6:0: E1132: Got multiple values for keyword argument 'that' in function call (repeated-keyword)
```
```


### Expected behavior

No warning

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.2 (default, Mar 27 2021, 14:26:03) 
[GCC 10.2.1 20210110]
```

**Comment by Pierre-Sassoulas** [comment]
I think this is not really a false positive, this code is tricky and should raise a warning. The warning should probably be on the dict call above instead of the function and be named something else though.

**Comment by alimuldal** [comment]
I don't agree that this code ought to trigger a warning, but even if you think it should, the warning message is clearly incorrect - it's not true that we're passing multiple values for `that`.

**Comment by Pierre-Sassoulas** [comment]
Yes, I imagine something like:
```
/tmp/test_module.py:6:0: E1132: Keyword 'that' implicitly replace existing value in 'kwargs' (implicitely-replaced-keyword)
```
What do you think ?

**Comment by alimuldal** [comment]
I disagree that this should be a warning - this is quite a common and useful pattern for creating a copy of a dict with some of the values replaced.

It's equivalent to:
```python
new_kwargs = {**kwargs, 'that': 3}
my_func(**new_kwargs)
```
which doesn't elicit a pylint warning. Do you think that pylint should treat `{**kwargs, 'that': 3}` and `dict(kwargs, that=3)` differently, and if so, why?

**Comment by alimuldal** [comment]
Hi, any more thoughts on this?

**Comment by Pierre-Sassoulas** [comment]
The discussion can continue here if someone is interested in fixing this and have an opinion about the subject.

**Comment by DanielNoord** [comment]
I think this is indeed an incorrect warning. Note that:
```python
def my_func(this, that):
    return this, that


kwargs = dict(this=1, that=2)
new_kwargs = dict(kwargs, that=3)
print(new_kwargs)
my_func(**new_kwargs)
```

Produces:
```console
❯ python test.py
{'this': 1, 'that': 3}
```

The `that` gets overwritten when `new_kwargs` is created, so there is no double value being passed to `my_func`.
